### ISSUE TITLE
CA2119 does not check the declaring type's accessibility or constructor accessibility

### ISSUE THREAD

**Issue by pakdev** [dotnet/roslyn-analyzers#1086]
#### Analyzer package

Microsoft.QualityGuidelines.Analyzers

#### Analyzer

SealMethodsThatSatisfyPrivateInterfaces

#### Repro steps

```cs
internal interface IFace
{
    void M();
}

// Declaring type only accessible to this assembly
internal class C : IFace
{
    public virtual void M()
    {
    }
}

// Declaring type can only be instantiated in this assembly
public class D : IFace
{
    internal D()
    {
    }

    public virtual void M()
    {
    }
}
```

#### Expected behavior

Should not fire when the declaring type's accessibility is ```internal``` or the declaring type's constructors aren't ```public```. This matches [FxCop's recommendations](https://msdn.microsoft.com/en-us/library/ms182313.aspx) on how to fix violations.

#### Actual behavior

Fires

**Comment by alejandro5042** [comment]
I have a proposed fix to this issue here: https://github.com/dotnet/roslyn-analyzers/pull/1136

**Comment by mavasani** [comment]
The fix has been merged. Thanks @alejandro5042
